bindings/matrix-sdk-ffi/src/encryption.rs

```diff
@@ -411,52 +412,51 @@ impl Encryption {
     }
 
-    /// Get the E2EE identity of a user from the crypto store.
+    /// Get the E2EE identity of a user.
     ///
-    /// Usually, we only have the E2EE identity of a user locally if the user
-    /// is tracked, meaning that we are both members of the same encrypted room.
-    ///
-    /// To get the E2EE identity of a user even if it is not available locally
-    /// use `request_user_identity()`.
-    ///
-    /// # Arguments
-    ///
-    /// * `user_id` - The unique id of the user that the identity belongs to.
-    ///
-    /// Returns a `UserIdentity` if one is found and the crypto store
-    /// didn't throw an error.
-    ///
-    /// This will always return None if the client hasn't been logged in.
-    pub async fn get_user_identity(
-        &self,
-        user_id: String,
-    ) -> Result<Option<Arc<UserIdentity>>, ClientError> {
-        let identity = self.inner.get_user_identity(user_id.as_str().try_into()?).await?;
-        Ok(identity.map(|i| Arc::new(UserIdentity { inner: i })))
-    }
-
-    /// Get the E2EE identity of a user from the homeserver.
-    ///
-    /// The E2EE identity returned is always guaranteed to be up-to-date. If the
-    /// E2EE identity is not found, it should mean that the user did not set
-    /// up cross-signing.
-    ///
-    /// If you want the E2EE identity of a user without making a request to the
-    /// homeserver, use `get_user_identity()` instead.
+    /// This method always tries to fetch the identity from the store, which we
+    /// only have if the user is tracked, meaning that we are both members
+    /// of the same encrypted room. Optionally it can request the user from the
+    /// homeserver if not found locally.
     ///
     /// # Arguments
     ///
     /// * `user_id` - The ID of the user that the identity belongs to.
     ///
+    /// * `request_if_needed` - If true, the identity will be requested from the
+    ///   homeserver if it is not found in the crypto store. When `true` the
+    ///   identity returned is guaranteed to be up-to-date.
+    ///
     /// Returns a `UserIdentity` if one is found. Returns an error if there
     /// was an issue with the crypto store or with the request to the
     /// homeserver.
     ///
     /// This will always return `None` if the client hasn't been logged in.
-    pub async fn request_user_identity(
+    pub async fn user_identity(
         &self,
         user_id: String,
+        request_from_homeserver_if_needed: bool,
     ) -> Result<Option<Arc<UserIdentity>>, ClientError> {
+        match self.inner.get_user_identity(user_id.as_str().try_into()?).await {
+            Ok(Some(identity)) => {
+                return Ok(Some(Arc::new(UserIdentity { inner: identity })));
+            }
+            Ok(None) => {
+                info!("No identity found in the store.");
+                if !request_from_homeserver_if_needed {
+                    return Ok(None);
+                }
+            }
+            Err(e) => {
```

Comment:
nit: please use `err` or `error` for this variable's name, one-letter named variables are bleh :P

Reply:
https://github.com/matrix-org/matrix-rust-sdk/commit/dce7297fcc5eef8f9c2a97d7549e12f4aba483ed